### PR TITLE
Updates to add optional metric labels for 'Controller' and 'Action'

### PR DESCRIPTION
**Pull request by billpieper**
Per Issue #44 

Add options that can be leveraged to enable `Controller` and/or `Action` labels for the metric

### CODE REVIEW

File: src/HttpRequestDurationsMiddlewareBuilderExtensions.cs [hunk 1]
```diff
@@ -29,6 +29,12 @@ public static IApplicationBuilder UsePrometheusRequestDurations(this IApplicatio
                 ??= (ICollectorRegistry)app.ApplicationServices.GetService(typeof(ICollectorRegistry))
                     ?? Metrics.DefaultCollectorRegistry;
 
+#if HasRoutes
+            // If we are going to set labels for Controller or Action -- then we need to make them readily available
+            if (options.IncludeController || options.IncludeAction)
+                app.UseMiddleware<CaptureRouteDataMiddleware>();
```
Comment (phnx47) [inline src/HttpRequestDurationsMiddlewareBuilderExtensions.cs:35]: Can you extract logic from `CaptureRouteDataMiddleware` and add to `HttpRequestDurationsMiddleware`?

Comment (phnx47) [reply src/HttpRequestDurationsMiddlewareBuilderExtensions.cs:35]: Maybe I need some details about `CaptureRouteDataMiddleware`, why it should be like this.

Comment (billpieper) [reply src/HttpRequestDurationsMiddlewareBuilderExtensions.cs:35]: I leveraged the same pattern to obtain and get `Controller` and `Action` added as labels as the prior prometheus-net sdk is doing.  Let me know if you want me to combine into the existing middleware...

Comment (phnx47) [reply src/HttpRequestDurationsMiddlewareBuilderExtensions.cs:35]: @billpieper Yes, I think that we need only 1 middleware `HttpRequestDurationsMiddleware`, some options can be enabled or disabled

**Review by phnx47 — APPROVED** [review]

**Pull request merged**